crossdock/server/server.py

```diff
@@ -130,79 +168,88 @@ def handle_trace_request(self, http_request, trace_request, span_handler, respon
         if span_handler:
```

Comment:
I suggest accepting `downstream`, not `trace_request`, because the latter is being overloaded by both start and join requests that happened to share `downstream` property. But in practice we don't care about top level request here, only the downstream.